core/src/main/scala/pureconfig/package.scala

```diff
@@ -143,9 +166,13 @@ package object pureconfig {
    */
   @throws[ConfigReaderException[_]]
   def loadConfigOrThrow[Config](path: Path)(implicit conv: ConfigConvert[Config], ct: ClassTag[Config]): Config = {
-    ConfigFactory.invalidateCaches()
-    val config = ConfigFactory.load(ConfigFactory.parseFile(path.toFile))
-    getResultOrThrow[Config](loadConfig[Config](config)(conv))
+    val errorOrConfig =
+      for {
+        _ <- invalidateCaches().right
+        rawConfig <- loadFile(path).right
+        config <- loadConfig[Config](rawConfig)(conv).right
+      } yield config
+    getResultOrThrow[Config](errorOrConfig)
```

Comment:
Separate one but in this release. Let's do the review of this PR to manage exceptions from typesafe API and use the PR that solves #162 to fix the API. I prefer to have two commits in the git history tbh.